### PR TITLE
Upgrade mongoose jid

### PR DESCRIPTION
**Pull request by NelsonVides**
Using code from https://github.com/esl/mongoose_jid/pull/7. ~Needs still releasing the new mongoose_jid to hex and cleaning the rebar.lock files accordingly.~

As described in that other PR:
> Remove fields that should actually never be used. The idea is that the
non-normalised parts MUST actually have never been used for comparison,
and are therefore not really needed. If only for printing and using them
in the to and from attributes of a stanza, clients SHOULD also
stringprep anyway, so the fact that the server might provide
non-normalised strings might be a problem for clients who're not
enforced to normalise them.
I've also been verifying how many other XMPP implementations tackle this
issue, and I've seen many other servers, and also a bunch of client
frameworks, simply capturing the parts and normalising them, and
building data structures that keep only their normalised pieces.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/103758) / small_tests / 1f90b48c99a670e72eb0885094d3cb641b2446da
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/103758/small_tests.23.1)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/103824) / small_tests / 1f2bb943388e93a5c3cbe39cd7c4b75ac9f0c332
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/103824/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/103824/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-31_10.07.09/lib.mongooseim.logs/run.2022-01-31_10.07.09/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/103863) / small_tests / d46ff515a3904821642364af36abc7872dccc5ba
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/103863/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/103863/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-31_10.27.32/lib.mongooseim.logs/run.2022-01-31_10.27.33/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/103931) / small_tests / e3d05a9200472c900e0a3d67b58b37b08474ce41
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/103931/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/103931/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-01-31_13.10.35/lib.mongooseim.logs/run.2022-01-31_13.10.36/suite.log.html)

File: rebar.config [hunk 1]
```diff
@@ -52,7 +52,8 @@
   {base16, "2.0.0"},
   {exml, "3.0.5", {pkg, hexml}},
   {jiffy, "1.0.8"},
-  {jid, "1.0.0", {pkg, mongoose_jid}},
+  % {jid, "1.0.0", {pkg, mongoose_jid}},
```
Comment (chrzaszcz) [inline rebar.config:55]: I guess this is temporary?

File: src/jingle_sip/jingle_sip_helper.erl [hunk 2]
```diff
@@ -50,7 +50,7 @@ maybe_rewrite_to_phone(Acc) ->
     ToRewrite = gen_mod:get_module_opt(Server, mod_jingle_sip, username_to_phone, []),
     case lists:keyfind(ToUser, 1, ToRewrite) of
         {ToUser, PhoneNumber} ->
-            JID#jid{user = PhoneNumber, luser = PhoneNumber};
+            JID#jid{luser = PhoneNumber};
```
Comment (chrzaszcz) [inline src/jingle_sip/jingle_sip_helper.erl:53]: Why change this? Won't they (`luser` and `user`) get out of sync?

File: src/muc_light/mod_muc_light_codec_modern.erl [hunk 3]
```diff
@@ -356,8 +356,7 @@ encode_iq({set, #create{} = Create, UniqueRequested},
     %% IQ reply "from"
     %% Sent from service JID when unique room was requested
     {ResFromJID, ResFromBin} = case UniqueRequested of
-                                   true -> {#jid{ server = RoomJID#jid.lserver,
-                                                  lserver = RoomJID#jid.lserver },
+                                   true -> {#jid{lserver = RoomJID#jid.lserver},
```
Comment (chrzaszcz) [inline src/muc_light/mod_muc_light_codec_modern.erl:359]: Same here, wouldn't they get inconsistent?

File: test/json_formatter_SUITE.erl [hunk 4]
```diff
@@ -361,10 +361,10 @@ example_acc(Body) ->
                               [{<<"type">>,<<"chat">>},{<<"id">>,<<"1111">>}],
                               [{xmlel,<<"body">>,[],
                                 [{xmlcdata,Body}]}]},
-                  from_jid => {jid,<<"userA">>,<<"localhost">>,<<>>,<<"usera">>,<<"localhost">>,<<>>},
```
Comment (chrzaszcz) [inline test/json_formatter_SUITE.erl:364]: Could you explain this change? I can't see where it's coming from.

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general. It is difficult to understand some of the changes though, especially the changes of user, server and resource to their respective lowercase equivalents. What was the rule? I added a few comments as well.

**Comment by NelsonVides** [comment]
> It looks good in general. however, it's difficult to tell how all the changes would affect the system, I guess that we might see the lowercase versions in some stanzas generated by the system?

I'm happy you reviewed this one, but this is a draft of a different jid change :D

What needs review first is this one, with plenty of refactorings and stuff but no functional change whatsoever: https://github.com/esl/mongoose_jid/pull/6

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/119676) / small_tests / 8c3087812cac649f111673889c4e573777e70cfc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/119676/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/119676/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-04-16_18.26.55/lib.mongooseim.logs/run.2022-04-16_18.26.56/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/120645) / small_tests / 98df6f7e0879facca005e697f30ecefd38b393df
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/120645/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/120645/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-04-26_13.40.52/lib.mongooseim.logs/run.2022-04-26_13.40.52/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/120802) / small_tests / 3f3f976c0cec13979959b76d11887a4a3c6b518d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/120802/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/120802/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-04-26_16.33.54/lib.mongooseim.logs/run.2022-04-26_16.33.54/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/120825) / small_tests / 36a8ca2fcf8733c5add3164b230a0535bbf92b19
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3525/120825/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3525/120825/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-04-26_16.36.37/lib.mongooseim.logs/run.2022-04-26_16.36.38/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good :+1:

**Pull request merged**